### ISSUE TITLE
Compilation errors for Message(s) with Timestamp(s)

### ISSUE THREAD

**Issue by alexrodriguez** [stephenh/ts-proto#49]
I have a proto message with a timestamp field, for which I want to generate an interface.
```
syntax = "proto3";
import "google/protobuf/timestamp.proto";

message Registration {
    string event_name = 1;
    google.protobuf.Timestamp date = 2;
}
```
I pass the options to `protoc` that the README suggests:
```
--ts_proto_opt=outputEncodeMethods=false,outputJsonMethods=false,outputClientImpl=false
```
and obtain the typescript interface.
```
/* eslint-disable */
import { Timestamp } from './google/protobuf/timestamp';

export interface Registration {
  eventName: string;
  date: Date | undefined;
}

function toTimestamp(date: Date): Timestamp {
  const seconds = date.getTime() / 1_000;
  const nanos = (date.getTime() % 1_000) * 1_000_000;
  return { seconds, nanos };
}

function fromTimestamp(t: Timestamp): Date {
  let millis = t.seconds * 1_000;
  millis += t.nanos / 1_000_000;
  return new Date(millis);
}

function fromJsonTimestamp(o: any): Date {
  if (o instanceof Date) {
    return o;
  } else if (typeof o === "string") {
    return new Date(o);
  } else {
    return fromTimestamp(Timestamp.fromJSON(o));
  }
}
```
When I try to compile that code I get:
```
src/registration.ts:28:26 - error TS2693: 'Timestamp' only refers to a type, but is being used as a value here.

28     return fromTimestamp(Timestamp.fromJSON(o));
                            ~~~~~~~~~

Found 1 error.
```
What am I missing?

**Comment by stephenh** [comment]
@alexrodriguez ah, it looks like the code that creates the `fromJsonTimestamp` doesn't realize that you've disabled the JSON methods. That should be a pretty easy fix...give me a day/few to get to it.

If you're blocked and want to try to make a PR, that'd be great, let me know and I can give a few pointers about where/what to change.

**Comment by alexrodriguez** [comment]
Thanks for the reply. Point me in the right direction and I'll take a stab at a PR.

**Comment by stephenh** [comment]
Great!

Generally, try something like:

* Copy/paste the https://github.com/stephenh/ts-proto/tree/master/integration/simple-long directory but change the `parameters.txt` to have what you have, pick a name like maybe `only-types` (instead of `simple-long`)
* Run `./integration/codegen.sh` to update the new `only-types/build` output and hopefully show the compile error that you're getting
* Change https://github.com/stephenh/ts-proto/blob/master/src/main.ts#L147 to include something like `&& the from json methods are not false`.

**Comment by stephenh** [comment]
Thanks for the PR! I've merged it and pushed release 1.17.1 to npm with the fix.
